pkg/liquidity-mining/contracts/GaugeWorkingBalanceHelper.sol

```diff
@@ -0,0 +1,144 @@
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-interfaces/contracts/solidity-utils/openzeppelin/IERC20.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IVeDelegation.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+
+interface IVeDelegationProxy is IVeDelegation {
+    function getVotingEscrow() external view returns (IERC20);
+}
+
+interface IGauge {
+    function balanceOf(address user) external view returns (uint256);
+
+    function totalSupply() external view returns (uint256);
+
+    // solhint-disable func-name-mixedcase
+    function working_balances(address user) external view returns (uint256);
+
+    function working_supply() external view returns (uint256);
+    // solhint-enable func-name-mixedcase
+}
+
+/**
+ * @notice Get the current and projected `working_balance` (effective veBAL boosted balance) for a user on a gauge.
+ * @dev The `working_balance` can range between 40% and 100% of the nominal user balance on a gauge, depending on the
+ * veBAL "voting power" of the user. This value generally decays over time, but will increase with additional veBAL
+ * locking.
+ *
+ * Also compute the working balance ratios: balance/supply, which is more informative when deciding whether
+ * it would be advantageous for the user to checkpoint a gauge.
+ */
+contract GaugeWorkingBalanceHelper {
+    using FixedPoint for uint256;
+
+    // 40% (minimum balance, with no veBAL)
+    uint256 private constant _TOKENLESS_PRODUCTION = 40e16;
+
+    IVeDelegationProxy private immutable _veDelegationProxy;
+    IERC20 private immutable _veBAL;
+
+    // The veBAL user balance always comes from the proxy (on L1 and L2), but versions deployed to some
+    // networks require the total supply of veBAL to come from the VotingEscrow instead.
+    bool private immutable _readTotalSupplyFromVE;
+
+    constructor(IVeDelegationProxy veDelegationProxy, bool readTotalSupplyFromVE) {
+        _veDelegationProxy = veDelegationProxy;
+        _veBAL = veDelegationProxy.getVotingEscrow();
+
+        _readTotalSupplyFromVE = readTotalSupplyFromVE;
+    }
+
+    /**
+     * @dev Returns the VotingEscrowDelegationProxy.
+     */
+    function getVotingEscrowDelegationProxy() external view returns (address) {
+        return address(_veDelegationProxy);
+    }
+
+    /**
+     * @dev Returns the VotingEscrow contract associated with the proxy.
+     */
+    function getVotingEscrow() external view returns (address) {
+        return address(_veBAL);
```

Comment:
This is a weird case, since it's defined as an ERC-20, but is actually a Curve contract (VotingEscrow). Some of our contracts (e.g., FeeDistributor) return it as IVotingEscrow (from us, not Curve). Others (e.g., VotingEscrowDelegationProxy) return it as an IERC20. I used that pattern.

So I suppose I could make it return an IERC20.